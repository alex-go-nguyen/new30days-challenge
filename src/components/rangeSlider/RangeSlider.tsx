import React, { useEffect, useRef, useState } from 'react';
import classNames from 'classnames/bind';
import styles from './RangeSlider.module.scss';

const cx = classNames.bind(styles);

export default function RangeSLider() {
    const parentElement = useRef<HTMLDivElement>(null);
    const [percent, setPercent] = useState(0);

    useEffect(() => {
        const handleMouseMove = (e: MouseEvent) => {
            const parentWidth = parentElement.current?.offsetWidth;

            const cursorX = e.clientX - (parentElement.current?.getBoundingClientRect().left || 0);

            const percent = Math.floor((cursorX / (parentWidth || 1)) * 100);

            setPercent(percent <= 100 ? percent : 100);
        };

        parentElement.current?.addEventListener('mousemove', handleMouseMove);

        return () => {
            parentElement.current?.removeEventListener('mousemove', handleMouseMove);
        };
    }, []);

    return (
        <div className={cx('container')} ref={parentElement}>
            <div className={cx('bar')} style={{ width: `${percent}%` }}>
                {percent}%
            </div>
        </div>
    );
}
