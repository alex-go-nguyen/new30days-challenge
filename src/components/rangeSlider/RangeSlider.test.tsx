import React from 'react';
import renderer from 'react-test-renderer';
import RangeSlider from './RangeSlider';

describe('RangeSlider', () => {
    const component = renderer.create(<RangeSlider />);
    let tree = component.toJSON();
    test('should render correctly', () => {
        expect(tree).toMatchSnapshot();
    });
});
