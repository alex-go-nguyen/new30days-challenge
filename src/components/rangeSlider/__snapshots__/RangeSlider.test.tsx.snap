// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="bar"
    style={
      {
        "width": "0%",
      }
    }
  >
    0
    %
  </div>
</div>
`;
